docs/spec/connection-handshake/L1_2.md

```diff
@@ -0,0 +1,264 @@
+
+
+###### Problem statement and guarantees
+A connection handshake allows, briefly, that two processes agree on a consistent view of their state, and each process allocates a new connection associated with this state.
+We identify the following requirements that the connection handshake should guarantee:
+
+- **[CH/Seq/1/Agreement]** If the initiator finishes the handshake, establishing a new connection `ci` with the responder, and the responder finishes the handshake obtaining the connection `cr`, then `ci == cr`.
+In other words, the two ends of the connection (one at the initiator and the other at the responder) are consistent.
+
+
+- **[CH/Seq/2/Termination]** Eventually, the two processes should both finish the handshake, establishing a connection.
+
+Requirement **[CH/Seq/1/Agreement]** represents the basic safety guarantee, while **[CH/Seq/2/Termination]** is the basic liveness guarantee we would like to have.
+
+The diagram below sketches the high-level problem statement.
+
+```
+                        IBC Connection
+     +-----------+    Handshake Protocol    +-----------+
+     |           |<-------- step 1 -------->|           |
+     | Initiator |    (execute handshake)   | Responder |
+     |           |                          |           |
+     |          _| (connection established) |_          |
+     |         |c|<------------------------>|c|         |
+     |          ▔|                          |▔          |
+     +-----------+                          +-----------+
+```
+
+
+## L2. Protocol specification / protocol view
+
+In this section we attempt to make the problem specification more concrete.
+We define what the processes are, their fault-model, as well as communication and synchrony assumptions.
+
+### 2.1 System model specification
+
+#### System model
+
+A **chain** is a state machine.
+Each chain may be replicated, i.e., consisting of a replicated state machine (RSM), or it may be a standalone process.
+In practice, there are certain requirements which each chain must satisfy (e.g., see [[ICS 024](#references)]).
+At this level of specification, however, we can assume each chain is a single **process**.
+
+The connection handshake protocol is asymmetric, meaning that the two chains that interact play different roles within the protocol.
+We take advantage of this asymmetry and explicitly label one of the chains with `I` (the initiating, or source, process), and the other with `R` (the responder, or destination, process).
+
+On top of each chain, one or multiple **applications** may be running (e.g., payment system or token exchanges, with their associated smart contracts).
+In the context of the connection handshake problem, it is an application that invokes the connection handshake protocol, and provides the necessary input for executing this protocol (we give more details on the invocation & input in section [2.2 Protocol](#22-protocol) below).
+
```

Comment:
Maybe it is better to use the spec term of `actor` here.
 `An actor, or a user (used interchangeably), is an entity interacting with the IBC protocol. An actor can be a human end-user, a module or smart contract running on a blockchain, or an off-chain relayer process capable of signing transactions.`
It is possible that the connection is initiated by an end-user or relayer, and the handshake completed by the relayer.